blueflood-core/src/test/java/com/rackspacecloud/blueflood/service/ScheduleContextAreKeysRunningTest.java

```diff
@@ -0,0 +1,212 @@
+
+        // given
+        ctx.update(currentTime - 2, shard);
+        ctx.scheduleEligibleSlots(1, 7200000);
+
+        int otherSlot = Granularity.MIN_20.slot(currentTime);
+        SlotKey otherSlotkey = SlotKey.of(Granularity.MIN_20, otherSlot, shard);
+
+        // precondition
+        assertEquals(1, ctx.getScheduledCount());
+        assertEquals(0, ctx.getRunningCount());
+
+        // when
+        boolean areKeysRunning = ctx.areChildKeysOrSelfKeyScheduledOrRunning(otherSlotkey);
+
+        // then
+        assertTrue(areKeysRunning);
+    }
+
+    @Test
+    public void unrelatedSlotScheduledReturnsFalse() {
+
+        // given
+        ctx.update(currentTime - 2, shard);
+        ctx.scheduleEligibleSlots(1, 7200000);
+
+        int otherSlot = Granularity.MIN_5.slot(currentTime - 5*60*1000); // check the previous slot from 5 minutes ago
+        SlotKey otherSlotkey = SlotKey.of(Granularity.MIN_5, otherSlot, shard);
+
+        // precondition
+        assertEquals(1, ctx.getScheduledCount());
+        assertEquals(0, ctx.getRunningCount());
+
+        // when
+        boolean areKeysRunning = ctx.areChildKeysOrSelfKeyScheduledOrRunning(otherSlotkey);
+
+        // then
+        assertFalse(areKeysRunning);
+    }
+
+    @Test
+    public void slotRunningReturnsTrue() {
+
+        // given
+        ctx.update(currentTime - 2, shard);
+        ctx.scheduleEligibleSlots(1, 7200000);
+
+        SlotKey runningSlot = ctx.getNextScheduled();
+
+        // precondition
+        assertEquals(0, ctx.getScheduledCount());
+        assertEquals(1, ctx.getRunningCount());
+        assertEquals(slotkey, runningSlot);
+
+        // when
+        boolean areKeysRunning = ctx.areChildKeysOrSelfKeyScheduledOrRunning(slotkey);
+
+        // then
+        assertTrue(areKeysRunning);
+    }
+
+    @Test
+    public void childSlotRunningReturnsTrue() {
+
+        // given
+        ctx.update(currentTime - 2, shard);
+        ctx.scheduleEligibleSlots(1, 7200000);
+
+        int otherSlot = Granularity.MIN_20.slot(currentTime);
+        SlotKey otherSlotkey = SlotKey.of(Granularity.MIN_20, otherSlot, shard);
+
+        SlotKey runningSlot = ctx.getNextScheduled();
+
+        // precondition
+        assertEquals(0, ctx.getScheduledCount());
+        assertEquals(1, ctx.getRunningCount());
+
+        // when
+        boolean areKeysRunning = ctx.areChildKeysOrSelfKeyScheduledOrRunning(otherSlotkey);
+
+        // then
+        assertTrue(areKeysRunning);
+    }
+
+    @Test
+    public void slotPushedBackReturnsTrue() {
+
+        // given
+        ctx.update(currentTime - 2, shard);
+        ctx.scheduleEligibleSlots(1, 7200000);
+
+        SlotKey runningSlot = ctx.getNextScheduled();
+
+        // precondition
+        assertEquals(0, ctx.getScheduledCount());
+        assertEquals(1, ctx.getRunningCount());
+
+        ctx.clearFromRunning(runningSlot);
+        ctx.pushBackToScheduled(runningSlot, false);
```

Comment:
I undestand your explanation but looking at it just gives an impression that these two calls happens one after another in a workflow which isnt the case. It might be appropriate to your test case as you documented here. I was just wondering what would `areChildKeysOrSelfKeyScheduledOrRunning` return when we only do pushBackToScheduled which is a valid scenario.